### PR TITLE
Implement contacts sorting feature

### LINKED ISSUE #71: Sort contacts
As an NUS computing student, I wish to sort my list of contacts on NetworkBook by graduation year, so that I can have an overview of my connections with other NUS students/alumni from a particular year.

### PR DESCRIPTION
**Pull request by xenosf**
Fixes #71

### CODE REVIEW

File: src/main/java/networkbook/model/person/PersonSortComparator.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+ * Provides an ordering for {@code Person} instances based on the values of a specific field.
+ */
+public class PersonSortComparator implements Comparator<Person> {
+
+    /** Comparator that does nothing. */
+    public static final Comparator<Person> EMPTY_COMPARATOR = (Person o1, Person o2) -> 0;
+
+    /** Comparator that puts empty optionals at the end of the list. */
+    public static final Comparator<Optional<?>> OPTIONAL_COMPARATOR = (Optional<?> o1, Optional<?> o2) -> {
+        if (o1.isPresent() && o2.isEmpty()) {
```
Comment (nknguyenhc) [inline src/main/java/networkbook/model/person/PersonSortComparator.java:20]: Can have an assertion here that either `o1` or `o2` must be empty.

File: src/main/java/networkbook/logic/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public SortCommand(PersonSortComparator comparator) {
+        this.comparator = comparator;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateSortedPersonList(comparator);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
```
Comment (nknguyenhc) [inline src/main/java/networkbook/logic/commands/SortCommand.java:38]: Do add test case to test the execution of this command!

**Review by nknguyenhc — CHANGES_REQUESTED** [review]
Looks great! Some small changes requested below.

File: src/main/java/networkbook/model/person/PersonSortComparator.java (hunk 1 above)
Comment (xenosf) [reply src/main/java/networkbook/model/person/PersonSortComparator.java:20]: o1 and o2 can both be present; both present and both empty will return 0 (when both are present/empty, neither will be ordered greater than the other)

Comment (nknguyenhc) [reply src/main/java/networkbook/model/person/PersonSortComparator.java:20]: ok I see

File: src/main/java/networkbook/logic/commands/SortCommand.java (hunk 2 above)
Comment (xenosf) [reply src/main/java/networkbook/logic/commands/SortCommand.java:38]: Ok done!

**Review by nknguyenhc — APPROVED** [review]
LGTM!

**Pull request merged**